src/panels/config/automation/action/types/ha-automation-action-if.ts

```diff
@@ -0,0 +1,104 @@
+import { CSSResultGroup, html, LitElement } from "lit";
+import { customElement, property } from "lit/decorators";
+import { fireEvent } from "../../../../../common/dom/fire_event";
+import { Action, IfAction } from "../../../../../data/script";
+import { haStyle } from "../../../../../resources/styles";
+import type { HomeAssistant } from "../../../../../types";
+import type { Condition } from "../../../../lovelace/common/validate-condition";
+import "../ha-automation-action";
+import "../../../../../components/ha-textfield";
+import type { ActionElement } from "../ha-automation-action-row";
+
+@customElement("ha-automation-action-if")
+export class HaIfAction extends LitElement implements ActionElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @property({ attribute: false }) public action!: IfAction;
+
+  public static get defaultConfig() {
+    return { if: [], then: [] };
+  }
+
+  protected render() {
+    const action = this.action;
+
+    return html`
+      <h3>
+        ${this.hass.localize(
+          "ui.panel.config.automation.editor.actions.type.if.if"
+        )}:
```

Comment:
```suggestion
        )}*:
```

Reply:
So, don't mind if I ask, what does this do? It makes it a generator?

Reply:
To show it's required